sql/persisted_variable.cc

```diff
@@ -768,13 +792,29 @@ bool Persisted_variables_cache::set_persist_options(bool plugin_options) {
         res = new (thd->mem_root)
             Item_int(iter->value.c_str(), (uint)iter->value.length());
         break;
-      case SHOW_CHAR:
       case SHOW_LEX_STRING:
-      case SHOW_BOOL:
-      case SHOW_MY_BOOL:
         res = new (thd->mem_root) Item_string(
             iter->value.c_str(), iter->value.length(), &my_charset_utf8mb4_bin);
         break;
+      case SHOW_CHAR: {
+        auto *plugin_var = sysvar->cast_pluginvar();
+        if ((plugin_var != nullptr &&
+             plugin_var->check_type(PLUGIN_VAR_ENUM)) ||
+            /* The only way to detect enum */
+            dynamic_cast<Sys_var_enum *>(sysvar) ||
+            dynamic_cast<Sys_var_multi_enum *>(sysvar)) {
+          res = create_item(thd, iter->value);
+        } else {
+          res = new (thd->mem_root)
+              Item_string(iter->value.c_str(), iter->value.length(),
+                          &my_charset_utf8mb4_bin);
+        }
+        break;
+      }
```

Comment:
is this change done to handle ENUMs? Item_string() doesn't work ENUMs? How did it work before? Please try to add comments to the new code. Don't explain what the code is trying to do, explain what is achieving, or why we need this change?

Reply:
This was removed according to Kamil's suggestion

Reply:
Ok, You didn't answer the question :) but I understood from result file. So enums were written as numbers in persist file. With your change, it is written as string.

Is it possible to seprate out this  enum/bool "value" to string change?

Reply:
Sorry, I thought that is not actual as code was removed... This conversion is done in sys_var_pluginvar::persist_only_to_string(). In general it should be possible to continue writing ENUMs as numbers in one change and do a separate change which will change persist_only_to_string() behavior for ENUMs. This also will affect some tests. I can try to separate that but what benefit will that give us?

Reply:
Ok to keep it for now.. since you already wrote it and tested.

1. limit the scope to actual bug will help reviewer and then if we revert we end up reverting only the actual code 
2. Easier to test
3. Lesser scope == less chance for errors and the fix could be approved faster.